pallets/nfts/src/lib.rs

```diff
@@ -1938,6 +2007,192 @@ pub mod pallet {
+		/// - `delegate`: The account that is going to lose their approval.
+		///
+		/// Emits `ApprovalCancelled` on success.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(52)]
+		#[pallet::weight(T::WeightInfo::cancel_collection_approval())]
+		pub fn cancel_collection_approval(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			delegate: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			let delegate = T::Lookup::lookup(delegate)?;
+			Self::do_cancel_collection_approval(origin, collection, delegate)
+		}
+
+		/// Force-cancel a collection approval granted by `owner` account.
+		///
+		/// Origin must be `ForceOrigin`.
+		///
+		/// Arguments:
+		/// - `owner`: The owner of the approval to be force-cancelled by the `origin`.
+		/// - `collection`: The collection whose approval will be cancelled.
+		/// - `delegate`: The account that is going to lose their approval.
+		///
+		/// Emits `ApprovalCancelled` on success.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(53)]
+		#[pallet::weight(T::WeightInfo::force_cancel_collection_approval())]
+		pub fn force_cancel_collection_approval(
+			origin: OriginFor<T>,
+			owner: AccountIdLookupOf<T>,
+			collection: T::CollectionId,
+			delegate: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			let maybe_check_origin = T::ForceOrigin::try_origin(origin)
+				.map(|_| None)
+				.or_else(|origin| ensure_signed(origin).map(Some).map_err(DispatchError::from))?;
+			let delegate = T::Lookup::lookup(delegate)?;
+			let owner = T::Lookup::lookup(owner)?;
+
+			if let Some(check_origin) = maybe_check_origin {
+				ensure!(
+					Self::is_admin(&collection, &check_origin) ||
+						Self::is_issuer(&collection, &check_origin),
+					Error::<T, I>::NoPermission
+				);
+			}
+			Self::do_cancel_collection_approval(owner, collection, delegate)
+		}
+
+		/// Cancel all collection approvals, up to a specified limit.
+		///
+		/// Origin must be Signed.
+		///
+		/// Arguments:
+		/// - `collection`: The collection whose approvals will be cleared.
+		/// - `limit`: The amount of collection approvals that will be cleared.
+		///
+		/// Emits `ApprovalsCancelled` on success.
+		///
+		/// Weight: `O(n)` where:
+		/// - `n = limit`
+		#[pallet::call_index(54)]
+		#[pallet::weight(T::WeightInfo::clear_collection_approvals(*limit))]
+		pub fn clear_collection_approvals(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			limit: u32,
+		) -> DispatchResultWithPostInfo {
+			let origin = ensure_signed(origin)
+				.map_err(|e| e.with_weight(T::WeightInfo::clear_collection_approvals(0)))?;
+			let removed_approvals = Self::do_clear_collection_approvals(origin, collection, limit)?;
+			Ok(Some(T::WeightInfo::clear_collection_approvals(removed_approvals)).into())
+		}
+
+		/// Force-cancel all collection approvals granted by `owner` account, up to a specified
+		/// limit.
+		///
+		/// Origin must be `ForceOrigin`.
```

Comment:
also needs comment to be updated to include admin.

Reply:
resolved due to https://github.com/r0gue-io/pop-node/pull/387#discussion_r1909116618